tests/math/test_is_independent.py

```diff
@@ -0,0 +1,548 @@
+
+    dev = qml.device("default.qubit", wires=1)
+
+    @qml.qnode(dev, interface=interface)
+    def const_circuit(x, y):
+        qml.RX(0.1, wires=0)
+        return qml.expval(qml.PauliZ(0))
+
+    constant_functions = [
+        const_circuit,
+        lambda x: np.arange(20).reshape((2, 5, 2)),
+        lambda x: (np.ones(3), -0.1),
+        qml.jacobian(lambda x, y: 4 * x - 2.1 * y, argnum=[0, 1]),
+    ]
+
+    args_constant = [
+        (0.1, np.array([-2.1, 0.1])),
+        (1.2,),
+        (np.ones((2, 3)),),
+        (np.ones((3, 8)) * 0.1, -0.2 * np.ones((3, 8))),
+    ]
+
+    @qml.qnode(dev, interface=interface)
+    def dependent_circuit(x, y, z):
+        qml.RX(0.1, wires=0)
+        qml.RY(y / 2, wires=0)
+        qml.RZ(qml.math.sin(z), wires=0)
+        return qml.expval(qml.PauliX(0))
+
+    dependent_functions = [
+        dependent_circuit,
+        np.array,
+        lambda x: np.array(x * 0.0),
+        lambda x: (1 + qml.math.tanh(1000 * x)) / 2,
+        *dependent_lambdas,
+    ]
+
+    args_dependent = [
+        (0.1, np.array([-2.1, 0.1]), -0.9),
+        (-4.1,),
+        (-4.1,),
+        (np.ones((3, 8)) * 1.1,),
+        *args_dependent_lambdas,
+    ]
+
+    @pytest.mark.parametrize("func, args", zip(constant_functions, args_constant))
+    def test_independent(self, func, args):
+        """Tests that an independent function is correctly detected as such."""
+        assert is_independent(func, self.interface, args)
+
+    @pytest.mark.parametrize("func, args", zip(dependent_functions, args_dependent))
+    def test_dependent(self, func, args):
+        """Tests that a dependent function is correctly detected as such."""
+        assert not is_independent(func, self.interface, args)
+
+    @pytest.mark.xfail
+    @pytest.mark.parametrize("func, args", zip(overlooked_lambdas, args_overlooked_lambdas))
+    def test_overlooked_dependence(self, func, args):
+        """Test that particular functions that dependent on the input
+        are overlooked."""
+        assert not is_independent(func, self.interface, args)
+
+    def test_kwargs_are_considered(self):
+        """Tests that kwargs are taken into account when checking
+        independence of outputs."""
+        f = lambda x, kw=False: 0.1 * x if kw else 0.2
+        jac = qml.jacobian(f, argnum=0)
+        args = (0.2,)
+        assert is_independent(f, self.interface, args)
+        assert not is_independent(f, self.interface, args, {"kw": True})
+        assert is_independent(jac, self.interface, args, {"kw": True})
+
+
+if have_jax:
+
+    class TestIsIndependentJax:
+        """Tests for is_independent, which tests a function to be
+        independent of its inputs, using JAX."""
+
+        interface = "jax"
+
+        @pytest.mark.parametrize("num", [0, 1, 2])
+        @pytest.mark.parametrize(
+            "args",
+            [
+                (0.2,),
+                (1.1, 3.2, 0.2),
+                (np.array([[0, 9.2], [-1.2, 3.2]]),),
```

Comment:
Should these be `jnp.array`?

Reply:
Both would be fine. Like this we test the function with Jax and compatibility with numpy, which kind of increases coverage.